### PR TITLE
Submit/pylint add a few more warnings

### PR DESCRIPTION
**Pull request by dcbaker**
The only thing I've actually caught is a couple of `if len(container)`, but the rest of these are pretty useful as well.

### CODE REVIEW

File: .pylintrc [hunk 1]
```diff
@@ -6,4 +6,10 @@ score=no
 
 [MESSAGES CONTROL]
 disable=all
-enable=unreachable
\ No newline at end of file
+enable=
+    assert-on-tuple,
```
Comment (mensinda) [inline .pylintrc:10]: Does this even work?

```bash
find mesonbuild -name "*.py" -type f -exec grep 'assert(' {} \;
```

prints a ton of results.

Comment (dcbaker) [reply .pylintrc:10]: You probably need to feel for `assert(.*,` to find what it's actually searching for

Comment (mensinda) [reply .pylintrc:10]: Oh, right. I was thinking of the `superfluous-paren` warnings, nevermind.

**Comment by nirbheek** [comment]
The MSYS2 failures are fixed. A rebase should do it.

**Pull request merged**